test/Concurrency/Runtime/async_sequence.swift

```diff
@@ -0,0 +1,524 @@
+  do {
+    let result = try await [1, 2, 3].async
+      .compactMap { try await asynchronousThrowing($0) }.collect()
+    expectEqual(result, [1, 2, 3])
+  } catch {
+    expectUnreachable()
+  }
+}
+
+AsyncCompactMapTests.test("nil values with async throw") {
+  do {
+    let result = try await [1, nil, 2, nil, 3, nil].async
+      .compactMap { try await asynchronousThrowing($0) }.collect()
+    expectEqual(result, [1, 2, 3])
+  } catch {
+    expectUnreachable()
+  }
+}
+
+AsyncCompactMapTests.test("throwing") {
+  do {
+    _ = try await [1, nil, 2, nil, 3, nil].async
+      .compactMap { value throws -> Int? in
+        if value == 2 {
+          throw Failure()
+        }
+        return value
+      }
+      .collect()
+    expectUnreachable()
+  } catch {
+
+  }
+}
+
+var AsyncMapSequenceTests = TestSuite("AsyncMapSequence")
+
+AsyncMapSequenceTests.test("map values") {
+  let results = await [1, 2, 3].async.map { "\($0)" }.collect()
+  expectEqual(results, ["1", "2", "3"])
+}
+
+AsyncMapSequenceTests.test("map empty") {
+  let results = await [Int]().async.map { "\($0)" }.collect()
+  expectEqual(results, [])
+}
+
+AsyncMapSequenceTests.test("map throwing") {
+  let seq = [1, 2, 3].async
+    .map { value throws -> String in
+      if value == 2 {
+        throw Failure()
+      }
+      return "\(value)" 
+    }
+  var it = seq.makeAsyncIterator()
+  var results = [String]()
+  do {
+    while let value = try await it.next() {
+      results.append(value)
+    }
+    expectUnreachable()
+  } catch {
+    expectEqual(results, ["1"])
+  }
+}
+
+var AsyncFilterSequenceTests = TestSuite("AsyncFilterSequence")
+
+AsyncFilterSequenceTests.test("filter out one item") {
+  let results = await [1, 2, 3].async.filter { $0 % 2 != 0 } .collect()
+  expectEqual(results, [1, 3])
+}
+
+AsyncFilterSequenceTests.test("filter out one item throwing") {
+  do {
+    let results = try await [1, 2, 3].async
+      .filter { try throwing($0 % 2 != 0) }
+      .collect()
+    expectEqual(results, [1, 3])
+  } catch {
+    expectUnreachable()
+  }
+}
+
+AsyncFilterSequenceTests.test("filter out one item throwing async") {
+  do {
+    let results = try await [1, 2, 3].async
+      .filter { try await asynchronousThrowing($0 % 2 != 0) }
+      .collect()
+    expectEqual(results, [1, 3])
+  } catch {
+    expectUnreachable()
+  }
+}
+
+AsyncFilterSequenceTests.test("filter out two items") {
+  let results = await [1, 2, 3].async.filter { $0 % 2 == 0 } .collect()
+  expectEqual(results, [2])
+}
+
+AsyncFilterSequenceTests.test("filter out all items") {
+  let results = await [1, 2, 3].async.filter { _ in return false } .collect()
+  expectEqual(results, [Int]())
+}
+
+AsyncFilterSequenceTests.test("filter out no items") {
+  let results = await [1, 2, 3].async.filter { _ in return true } .collect()
+  expectEqual(results, [1, 2, 3])
+}
+
+var AsyncDropWhileSequenceTests = TestSuite("AsyncDropWhileSequence")
```

Comment:
Need a throwing test in this section.

Reply:
same for the third and fourth test as before

Reply:
I see what you mean, actually throwing an error.. will add